tools/qemu_dev_start

```diff
@@ -130,33 +145,48 @@ parse_commandline "$@"
 handle_passed_args_count
 assign_positional_args 1 "${_positionals[@]}"
 
-
 #####################
 
 
 if [[ "$_arg_verbose" == "on" ]]; then
     set -x
 fi
 
-# LAN ifc. We may use a DHCP client, but we are forcing a fixed ip
-ip tuntap add name lime_tap0 mode tap
-ip addr add 10.13.0.99/24 dev lime_tap0
-ip link set lime_tap0 up
+BRIDGE_IFC="lime_br0"
+NODE_NUMBER="${_arg_node_number}"
+LAN_MAC="52:00:00:ab:cd:0${NODE_NUMBER}"
+WAN_MAC="52:00:00:ab:cd:a${NODE_NUMBER}"
+LAN_IFC="lime_tap${NODE_NUMBER}0"
+WAN_IFC="lime_tap${NODE_NUMBER}1"
+OWN_LAN_IP="10.13.0.2/16"
+
+
+if [ ! -e "/sys/class/net/$BRIDGE_IFC" ]; then
+    ip link add name "$BRIDGE_IFC" type bridge
+    ip link set "$BRIDGE_IFC" up
+    ip addr add "$OWN_LAN_IP" dev "$BRIDGE_IFC"
+fi
+
+# No IP for the tap LAN ifc
+ip tuntap add name "$LAN_IFC" mode tap
+ip link set "$LAN_IFC" up
+ip link set "$LAN_IFC" master "$BRIDGE_IFC"
+
 
 if [[ "$_arg_enable_wan" == "on" ]]; then
     # WAN ifc
-    ip tuntap add name lime_tap1 mode tap
-    ip addr add 172.99.0.1/24 dev lime_tap1
-    ip link set lime_tap1 up
+    ip tuntap add name "$WAN_IFC" mode tap
+    ip addr add 172.99.0.1/24 dev "$WAN_IFC"
+    ip link set "$WAN_IFC" up
 
     # DHCP server for WAN ifc
-    dnsmasq -F 172.99.0.100,172.99.0.100 --dhcp-option=3,172.99.0.1 -i lime_tap1 --dhcp-authoritative --log-dhcp
+    dnsmasq -F 172.99.0.100,172.99.0.100 --dhcp-option=3,172.99.0.1 -i "$WAN_IFC" --dhcp-authoritative --log-dhcp
 
     # enable forwarding and NAT
     echo 1 > /proc/sys/net/ipv4/ip_forward
     iptables -t nat -A POSTROUTING -o wlan0 -j MASQUERADE
     iptables -A FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-    iptables -A FORWARD -i  lime_tap1  -o wlan0 -j ACCEPT
+    iptables -A FORWARD -i  "$WAN_IFC"  -o wlan0 -j ACCEPT
```

Comment:
It assumes that the wireless interface is called wlan0 (in my computer for example is wlp2s0), and that that one is the access to the internet.

Reply:
Yes, maybe we can use an program argument for this.

Reply:
The WAN functionality is fragile also beacause of how the dhcp server is implemented (some computers have a dhcp server already binded to all the interfaces)

Reply:
Fixed :)